hs_core/views/utils.py

```diff
@@ -404,3 +404,278 @@ def show_relations_section(res_obj):
         return True
     return False
+
+
+def link_irods_file_to_django(resource, filename, size=0):
+    # link the newly created zip file to Django resource model
+    b_add_file = False
+    if resource:
+        if resource.resource_federation_path:
+            if not ResourceFile.objects.filter(object_id=resource.id,
+                                               fed_resource_file_name_or_path=filename).exists():
+                ResourceFile.objects.create(content_object=resource,
+                                            resource_file=None,
+                                            fed_resource_file_name_or_path=filename,
+                                            fed_resource_file_size=size)
+                b_add_file = True
+
+        elif not ResourceFile.objects.filter(object_id=resource.id,
+                                             resource_file=filename).exists():
+                ResourceFile.objects.create(content_object=resource,
+                                            resource_file=filename)
+                b_add_file = True
+
+        if b_add_file:
+            file_format_type = get_file_mime_type(filename)
+            if file_format_type not in [mime.value for mime in resource.metadata.formats.all()]:
+                resource.metadata.create_element('format', value=file_format_type)
+
+
+def link_irods_folder_to_django(resource, istorage, foldername, exclude=()):
+    """
+    Recursively Link irods folder and all files and sub-folders inside the folder to Django
+    Database after iRODS file and folder operations to get Django and iRODS in sync
+    :param resource: the BaseResource object representing a HydroShare resource
+    :param istorage: IrodsStorage object
+    :param foldername: the folder name
+    :param exclude: a tuple that includes file names to be excluded from linking under the folder;
+     default is empty meaning nothing is excluded.
+    :return:
+    """
+    if resource and istorage and foldername:
+        store = istorage.listdir(foldername)
+        # add the unzipped files into Django resource model
+        for file in store[1]:
+            if file not in exclude:
+                file_path = os.path.join(foldername, file)
+                size = istorage.size(file_path)
+                link_irods_file_to_django(resource, file_path, size)
+        for folder in store[0]:
+            link_irods_folder_to_django(resource,
+                                        istorage, os.path.join(foldername, folder), exclude)
+
+
+def rename_irods_file_or_folder_in_django(resource, src_name, tgt_name):
+    """
+    Rename file in Django DB after the file is renamed in Django side
+    :param resource: the BaseResource object representing a HydroShare resource
+    :param src_name: the file or folder full path name to be renamed
+    :param tgt_name: the file or folder full path name to be renamed to
+    :return:
+    """
+    if resource:
```

Comment:
Why check only one parameter and not the rest? In fact, I don't think this check is necessary.